script.js

```diff
@@ -136,18 +328,67 @@ my_widget_script =
         //LA calls this method with b_suppress_message and relies on your code to communicate issues to the user
         //Returning an empty array [] or NULL equals no error
         //TO DO write code specific to your form
-     
+        console.log('In is_valid:function')
         return this.parent_class.is_valid(b_suppress_message);
       },
-     
+
       is_edited:function () {
         //should return true if the form has been edited since it was loaded or since reset_edited was called
+        console.log('In is_edited:function')
         return this.parent_class.is_edited();
       },
-     
+
       reset_edited:function () {
         //typically called have a save
         //TO DO write code specific to your form
         return this.parent_class.reset_edited();
-      }
-    }
+      },
+
+      /*
+			Enable all records if the first record fields are correctly filled
+	  */
+      enable_all_records: function() {
+        console.log('**** In enable_all_records ****')
+        var observationsArray = ['2', '3', '4', '5', '6', '7', '8', '9', '10']
```

Comment:
These can be numbers, since `+` between a string and a number returns a string. Since these can be numbers you should just be using `for (var i = 2; i < N_ROWS + 1; i++)` to iterate over them